runtime/lua/vim/lsp/util.lua

```diff
@@ -105,12 +105,20 @@ function M.apply_text_edits(text_edits, bufnr)
     -- adjust start and end column for UTF-16 encoding of non-ASCII characters
     local start_row = e.range.start.line
     local start_col = e.range.start.character
-    local start_bline = api.nvim_buf_get_lines(bufnr, start_row, start_row+1, true)[1]
-    start_col = vim.str_byteindex(start_bline, start_col)
+    if start_col > 0 then
+      local start_bline = api.nvim_buf_get_lines(bufnr, start_row, start_row+1, false)
+      if #start_bline > 0 then
+        start_col = vim.str_byteindex(start_bline[1], start_col)
+      end
+    end
```

Comment:
Perhaps this section could be wrapped up in another util function and shared between here and line 116 (since they're both doing the same thing). It seems easy to forget to handle these problems later (multi byte stuff, not enough lines, etc.). Should wrap them up in one spot.

Reply:
Done!